simulation/test_session.cpp

```diff
@@ -64,3 +64,51 @@ TORRENT_TEST(seed_mode)
 		});
 }
 
+TORRENT_TEST(ip_notifier_setting)
+{
+	int s_tick = 0;
+	bool working = false;
+
+	setup_swarm(1, swarm_test::upload
+		// add session
+		, [](lt::settings_pack& pack)
+		{
+			pack.set_int(settings_pack::tick_interval, 1000);
+			pack.set_int(settings_pack::alert_mask, alert::all_categories);
+		}
+		// add torrent
+		, [](lt::add_torrent_params& params) {}
+		// on alert
+		, [&s_tick, &working](lt::alert const* a, lt::session& ses)
+		{
+			std::string const msg = a->message();
+			if (msg.find("received error on_ip_change:") != std::string::npos)
+			{
+				TEST_CHECK(s_tick == 0 || s_tick == 2);
+				working = true;
```

Comment:
so this is expected to happen twice, right? this test only asserts that it happens at least once. perhaps "working" should be a counter.

Reply:
yes, it's better